File: Core/actor/actorconfigmessage.cpp
#include "actorconfigmessage.h"

ActorConfigMessage::ActorConfigMessage(QString actorGroupId, QString actorId, QMap<QString, QVariant> values, QObject *parent)
    : MessageBase(parent), m_actorGroupId(actorGroupId), m_actorId(actorId), m_values(values)
{

}

MessageBase::MESSAGE_TYPE ActorConfigMessage::getMessageType() {
    return MESSAGE_TYPE_ACTOR_CONFIG;
}

QString ActorConfigMessage::getFirstLevelId() {
    return m_actorGroupId;
}

QString ActorConfigMessage::getSecondLevelId() {
    return m_actorId;
}

QString ActorConfigMessage::actorGroupId() {
    return m_actorGroupId;
}

QString ActorConfigMessage::actorId() {
    return m_actorId;
}

QString ActorConfigMessage::fullId() {
    return m_actorGroupId + VALUE_SEPARATOR + m_actorId;
}

QMap<QString, QVariant> ActorConfigMessage::values() {
    return m_values;
}

File: AndroidApp/OshCore/src/main/java/com/osh/actor/ActorConfigMessage.java
package com.osh.actor;

import java.util.Map;

import com.osh.communication.MessageBase;

public class ActorConfigMessage extends MessageBase {
	
	private String actorGroupId;
	private String actorId;
	private Map<String, Object> values;
	
	public ActorConfigMessage(String actorGroupId, String actorId, Map<String, Object> values) {
		super();
		this.actorGroupId = actorGroupId;
		this.actorId = actorId;
		this.values = values;
	}

	@Override
	public MESSAGE_TYPE getMessageType() {
		return MESSAGE_TYPE.MESSAGE_TYPE_ACTOR_CONFIG;
	}

	@Override
	public String getFirstLevelId() {
		return actorGroupId;
	}

	@Override
	public String getSecondLevelId() {
		return actorId;
	}

	public String getActorGroupId() {
		return actorGroupId;
	}

	public String getActorId() {
		return actorId;
	}

	public Map<String, Object> getValues() {
		return values;
	}
	
}
